crates/bevy_app/src/main_schedule.rs

```diff
@@ -0,0 +1,127 @@
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct First;
+
+/// The schedule that contains work required to make [`Update`] logic.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PreUpdate;
+
+/// Runs state transitions.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct StateTransition;
+
+/// Runs the [`FixedUpdate`] schedule in a loop according until all relevant elapsed time has been "consumed".
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FixedUpdateLoop;
+
+/// The schedule that contains systems which only run after a fixed period of time has elapsed.
+///
+/// The exclusive `run_fixed_update_schedule` system runs this schedule.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FixedUpdate;
+
+/// The schedule that contains app logic.
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Update;
+
+/// The schedule that contains systems that respond to [`Update`].
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct PostUpdate;
+
+/// Runs last in the schedule
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Last;
+
+/// Defines the schedules to be run for the [`Main`] schedule, including
+/// their order.
+#[derive(Resource, Debug)]
+pub struct MainScheduleOrder {
+    labels: Vec<Box<dyn ScheduleLabel>>,
+}
+
+impl Default for MainScheduleOrder {
+    fn default() -> Self {
+        Self {
+            labels: vec![
+                Box::new(First),
+                Box::new(PreUpdate),
+                Box::new(StateTransition),
+                Box::new(FixedUpdateLoop),
+                Box::new(Update),
+                Box::new(PostUpdate),
+                Box::new(Last),
+            ],
+        }
+    }
+}
+
+impl MainScheduleOrder {
+    /// Adds the given `schedule` after the `after` schedule
+    pub fn insert_after(&mut self, after: impl ScheduleLabel, schedule: impl ScheduleLabel) {
+        let index = self
+            .labels
+            .iter()
+            .position(|current| (**current).eq(&after))
+            .unwrap_or_else(|| panic!("Expected {after:?} to exist"));
+        self.labels.insert(index + 1, Box::new(schedule));
+    }
+}
+
+impl Main {
+    /// A system that runs the "main schedule"
+    pub fn run_main(world: &mut World, mut run_at_least_once: Local<bool>) {
+        if !*run_at_least_once {
+            world.run_schedule(PreStartup);
+            world.run_schedule(Startup);
+            world.run_schedule(PostStartup);
+            *run_at_least_once = true;
+        }
+
+        world.resource_scope(|world, order: Mut<MainScheduleOrder>| {
+            for label in &order.labels {
+                world.run_schedule_ref(&**label);
+            }
+        });
+    }
+
+    /// Initializes the [`Main`] schedule and sub-schedules on the given `app`.
+    pub fn init(app: &mut App) {
```

Comment:
This kind of feels like something that should be done in a `Plugin`.

Reply:
Good instincts. I like it!